### DISCUSSION TITLE
Contribution activity is not marked with a green icon.

### DISCUSSION THREAD

**Discussion by PARKSUNJUNG** [community/community#31473] [Other Feature Feedback, Questions, & Ideas]
![image](https://user-images.githubusercontent.com/61115291/187653945-6cca7cef-61c0-48e6-ab5f-606275df4fae.png)

Commits to github but contribution activity is not marked with a green icon.

At first, even after committing, it was not recorded with a green icon.
It turns out that in my case, I was using Eclipse and Git connected, but there was no local PC email because I did not install Git on a separate local PC.
After installing Git on your local PC, when you set up an email with the following command, it is displayed as a green icon.
> git config --global user.email "useremail"
> git config --global user.name "username"

However, the history that has been committed so far still does not appear as a green icon.

I want all committed activities to be represented by a green icon.

Help!

**Comment by melvincwng** [comment]
@PARKSUNJUNG, Hi there, to ensure your contribution activity is marked with a green icon, please follow the steps below:

1. Ensure that you push your commits to the master/main branch of your code repository.
2. If your code repository is private, the commits activity won't be shown in your GitHub profile by default. To show your private contribution/commits in your GitHub Profile, please refer to this [GitHub Documentation](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/showing-your-private-contributions-and-achievements-on-your-profile).

Hope this answers your question 😊

**Comment by airtower-luna** [comment]
> However, the history that has been committed so far still does not appear as a green icon.

That's expected: Setting `user.email` sets the author email address that'll be included in *future* commits. It does not modify existing commits.

If you want to fix the author email address in past commits you'll have to [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). Note that this can cause trouble if you've already shared it with others, because everyone would need to *manually* accept the changed history. :warning: If you want to do it, there's a "Changing Email Addresses Globally" section at the bottom of the linked article that describes how.
